pkg/url/url.go

```diff
@@ -899,11 +924,55 @@ func Push(client *occlient.Client, kClient *kclient.Client, parameters PushParam
 					Port:   url.Port,
 					Secure: url.Secure,
 					Kind:   envinfo.ROUTE,
+					Path:   "/",
 				},
 			}
 		}
 	}
 
+	// iterate through endpoints defined in devfile
+	// add the url defination into urlLOCAL if it's not defined in env.yaml
+	if parameters.IsExperimentalModeEnabled && parameters.EndpointMap != nil {
+		for port, endpoint := range parameters.EndpointMap {
+			// should not create URL if Exposure is none or internal
+			if endpoint.Exposure == "none" || endpoint.Exposure == "internal" {
+				continue
+			}
+			exist := false
+			for _, envURL := range urlLOCAL {
+				if envURL.Spec.Port == int(port) {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				// create route against Openshift
+				if parameters.IsRouteSupported {
+					secure := false
+					if endpoint.Secure || endpoint.Protocol == "https" || endpoint.Protocol == "wss" {
+						secure = true
+					}
+					path := "/"
+					if endpoint.Path != "" {
+						path = endpoint.Path
+					}
+					name := strings.TrimSpace(util.GetDNS1123Name(strings.ToLower(endpoint.Name)))
+					urlLOCAL[name] = URL{
+						Spec: URLSpec{
+							Port:   int(port),
+							Secure: secure,
+							Kind:   envinfo.ROUTE,
+							Path:   path,
+						},
+					}
+				} else {
+					// display warning since Host info is missing
+					log.Warningf("Unable to create ingress, missing host information for Endpoint %v, please check instructions on URL creation (refer `odo url create --help`)\n", endpoint.Name)
```

Comment:
this should be printed after `Applying URL change`

Reply:
updated